### PR TITLE
Adding python3.8 testing

### PR DESCRIPTION
**Pull request by bsipocz**
The new python 3.8.0 is not yet available via conda, thus this cannot be added yet with the usual ci-helpers way.

However, no dependencies are available as wheels for python38 just yet, so in the meantime  this job would build them.
(or alternatively we could use, e.g. numpy dev nightly wheel)

### CODE REVIEW

**Comment by bsipocz** [comment]
See related upstream issue https://github.com/numpy/numpy/issues/14698

**Comment by bsipocz** [comment]
Well, it seems that some kind of numpy dev version is already natively included in the travis python 3.8-dev environment.
https://travis-ci.org/astropy/astropy/jobs/597935269#L258

**Comment by bsipocz** [comment]
I'm cc-ing @mhvk and @pllim whether they see any logical combination of this job with any of our existing jobs. Shall I make this our numpy-dev job?

**Comment by bsipocz** [comment]
(travis jobs that are not the python 3.8 one are cancelled)

**Comment by bsipocz** [comment]
The test failures are due to deprecation warnings triggered by mpl, however they have been already been fixed upstream, so I add them to the list for now to ignore. https://github.com/matplotlib/matplotlib/pull/15168

**Comment by mhvk** [comment]
Using it for `numpy-dev` seems reasonable.

File: astropy/conftest.py [hunk 1]
```diff
@@ -23,7 +23,12 @@
     # This is a workaround for the OpenSSL deprecation warning that comes from
     # the `requests` module. It only appears when both asdf and sphinx are
     # installed. This can be removed once pyopenssl 1.7.20+ is released.
-    modules_to_ignore_on_import=['requests'])
+    modules_to_ignore_on_import=['requests'],
+    warnings_to_ignore_by_pyver={
```
Comment (pllim) [inline astropy/conftest.py]: It is still unclear to me when to add it here and when to add it to

https://github.com/astropy/astropy/blob/4b7fcafa0b3b4e30d10ce9ffe60aece4a6ef3675/astropy/tests/helper.py#L137

**Comment by mhvk** [comment]
Agreed with @pllim - now that it is released, it should be a proper job. Could it just be the "Initial tests" run?

**Comment by bsipocz** [comment]
> Agreed with @pllim - now that it is released, it should be a proper job. Could it just be the "Initial tests" run?

not yet, but we can swap the current initial one to be 3.8 once everything is properly out, but just not yet (e.g we only have dev numpy for 3.8, and I wouldn't put that in as an initial test).

File: astropy/conftest.py (hunk 1 above)
Comment (bsipocz) [reply astropy/conftest.py]: I'm on the side of we should set the very minimum of it in `astropy/astropy/tests/helper.py`, so downstream won't rely on it (and be surprised when we clean that file). I'm pretty sure there'll be a mpl version before 3.8 becomes widely used, so this should be a rather short term workaround anyway.

**Pull request merged**